jobserv_worker.py

```diff
@@ -492,10 +492,28 @@ def _update_max_memory(rundef):
         rundef["max-mem-bytes"] = total
 
 
+def _block_metadata_service():
+    """Block access to GCP/AWS metadata instance service if configured."""
+    block = config["jobserv"].get("block-metadata-service")
+    if block:
+        with open("/proc/net/route") as f:
+            for line in f:
+                parts = line.split()
+                if parts[1].lower() == "fea9fea9":
+                    # 169.254.169.254 encoded
+                    flags = int(parts[3], base=8)
+                    if flags & 11:  # 11 - "unreachable route"
+                        log.warning("Route to cloud metadata service already blocked")
+                        return
+        log.warning("Blocking route to cloud metadata service")
+        subprocess.check_call(["ip", "route", "add", "unreachable", "169.254.169.254"])
```

Comment:
Don't we want to make the metadata service IP address configurable?

Reply:
no. its a well-known value like 127.0.0.1 for cloud provider

Reply:
it blocks a user from running something like `curl http://169.254.169.254/latest/meta-data/` in their CI job

Reply:
i also just saw i need to block ipv6 `[fd00:ec2::254]`

Reply:
accidentally pushed this to main. I did some cleanups. Let me know if you object and I'll fix.

Reply:
Whatever warks for you is fine.